packages/contracts-bedrock/src/L2/L1Block.sol

```diff
@@ -168,14 +206,105 @@ contract L1Block is ISemver, IGasToken {
+            Storage.setAddress(L1_STANDARD_BRIDGE_ADDRESS_SLOT, abi.decode(_value, (address)));
+        } else if (_type == Types.ConfigType.SEQUENCER_FEE_VAULT_CONFIG) {
+            Storage.setBytes32(SEQUENCER_FEE_VAULT_CONFIG_SLOT, abi.decode(_value, (bytes32)));
+        }
+    }
+
+    function getConfig(Types.ConfigType _type) public view virtual returns (bytes memory data_) {
+        if (_type == Types.ConfigType.GAS_PAYING_TOKEN) {
+            (address addr, uint8 decimals) = gasPayingToken();
+            data_ = abi.encode(
+                addr,
+                decimals,
+                GasPayingToken.sanitize(gasPayingTokenName()),
+                GasPayingToken.sanitize(gasPayingTokenSymbol())
+            );
+        } else if (_type == Types.ConfigType.BASE_FEE_VAULT_CONFIG) {
+            data_ = abi.encode(Storage.getBytes32(BASE_FEE_VAULT_CONFIG_SLOT));
+        } else if (_type == Types.ConfigType.L1_ERC_721_BRIDGE_ADDRESS) {
+            data_ = abi.encode(Storage.getAddress(L1_ERC_721_BRIDGE_ADDRESS_SLOT));
+        } else if (_type == Types.ConfigType.REMOTE_CHAIN_ID) {
+            data_ = abi.encode(Storage.getUint(REMOTE_CHAIN_ID_SLOT));
+        } else if (_type == Types.ConfigType.L1_CROSS_DOMAIN_MESSENGER_ADDRESS) {
+            data_ = abi.encode(Storage.getAddress(L1_CROSS_DOMAIN_MESSENGER_ADDRESS_SLOT));
+        } else if (_type == Types.ConfigType.L1_STANDARD_BRIDGE_ADDRESS) {
+            data_ = abi.encode(Storage.getAddress(L1_STANDARD_BRIDGE_ADDRESS_SLOT));
+        } else if (_type == Types.ConfigType.SEQUENCER_FEE_VAULT_CONFIG) {
+            data_ = abi.encode(Storage.getBytes32(SEQUENCER_FEE_VAULT_CONFIG_SLOT));
+        } else if (_type == Types.ConfigType.L1_FEE_VAULT_CONFIG) {
+            data_ = abi.encode(Storage.getBytes32(L1_FEE_VAULT_CONFIG_SLOT));
+        }
+    }
+
     /// @notice Sets the gas paying token for the L2 system. Can only be called by the special
-    ///         depositor account. This function is not called on every L2 block but instead
-    ///         only called by specially crafted L1 deposit transactions.
+    ///         depositor account, initiated by a deposit transaction from L1.
+    ///         This is a legacy setter that exists to give compabilitity with the legacy SystemConfig.
+    ///         Custom gas token can now be set using `setConfig`. This can be removed in the future.
     function setGasPayingToken(address _token, uint8 _decimals, bytes32 _name, bytes32 _symbol) external {
         if (msg.sender != DEPOSITOR_ACCOUNT()) revert NotDepositor();
 
         GasPayingToken.set({ _token: _token, _decimals: _decimals, _name: _name, _symbol: _symbol });
-
         emit GasPayingTokenSet({ token: _token, decimals: _decimals, name: _name, symbol: _symbol });
     }
+
+    /// @notice Sets the L1 block values for an Isthmus upgraded chain.
+    function setL1BlockValuesIsthmus() external {
```

Comment:
This is not a setter like `setL1BlockValues`. This is not called by the L1 attributes transaction. This is called once, for other functions that are called once during the upgrade we follow the standard of `setXXX` where xxx is the hardfork name

Reply:
got it.

Reply:
f4ddb59e9648